### ISSUE TITLE
[a11y]: Search - Test- The screen reader announces nothing when the toggle button is activated

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16635]
### Package

@carbon/react

### Browser

Safari

### Package version

v1.58.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

When the user clicks on the 'Toggle Expand' button, the search button above it expands and collapses, but this is not announced to the user.
Note: The 'Toggle Expand' and 'Search' buttons have the same functionality in this story 

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-search--test

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
This is the same scenario as this https://github.com/carbon-design-system/carbon/issues/14817.  
> This one is kinda trick due to this scenario:
> 
> The aria-expanded attribute is in the Search icon button and when we click to expand and show the search input, the focus is redirected to the input, skipping the aria-expanded to be read.
> Since we want to keep this focus functionality the expanded announce won't be possible.
> 
> But the confusion can be solved by using the aria-label or placeholder to indicate what is happening on the screen and they have to type.
> 
> Note: I tried to use aria-live, aria-busy and aria-atomic but due to the JS focus being placed on the input it will be skipped.

An announcement to made by SR indicating that focus is on a textInput which conveys the users that search bar is opened. Also to be noted that `Test` story for `Search` component  was removed [here](https://github.com/carbon-design-system/carbon/commit/7b24a1058770c0c60c8e13995bd235d8f6f16c79#diff-42193b06f2861d4ab793f95e43fab6618e8637d3fa0a5a19dc3614e2cdfc23e3) from repro
